### ISSUE TITLE
http caching could be improved

### ISSUE THREAD

**Issue by gugurete** [nasa-gibs/worldview#289]
tile server used by https://worldview.earthdata.nasa.gov has http caching disabled :
```
Request URL:https://gibs-b.earthdata.nasa.gov/wmts/epsg4326/best/wmts.cgi?layer=Coastlines&style=default&tilematrixset=250m&Service=WMTS&Request=GetTile&Version=1.0.0&Format=image%2Fpng&TileMatrix=5&TileCol=16&TileRow=3
Request Method:GET
Status Code:200 OK
Remote Address:198.118.199.5:443
Referrer Policy:no-referrer-when-downgrade
```

```
Access-Control-Allow-Origin:*
Cache-Control:no-cache, no-store, must-revalidate
Connection:Keep-Alive
Content-Length:1096
Content-Type:image/png
Date:Thu, 18 May 2017 00:08:39 GMT
Expires:Thu, 18 May 2017 00:08:39 GMT
Keep-Alive:timeout=5, max=100
Pragma:no-cache
Server:Apache
Strict-Transport-Security:max-age=31536000
X-Frame-Options:SAMEORIGIN
```

setting a http header like **cache-control:public, max-age=31536000** for static image data as in the case of map tiles will improve performance dramatically

**Comment by rboller** [comment]
Thanks for the suggestion, we've gotten similar requests in the past though it's not entirely trivial in this case.  The first hurdle is that map tiles for "today" and "yesterday" get frequently updated and we wouldn't want the browser to show a mish-mash of old and new tiles.  The second is that, at least when we looked into it a few years ago, Worldview relies on "fresh" tiles for certain features.  We'll plan to take another look to see if that's still the case, though.

**Comment by bairgon** [comment]
You could set a short timeout for the cache, say 10 minutes, which would be of some benefit in reducing traffic. The browser will automatically validate the data with the server, getting a 304 response (not modified) if the file has not changed. This requires a server round-trip but doesn't send the data again.

For more information see https://devcenter.heroku.com/articles/increasing-application-performance-with-http-cache-headers . The timeout would be 600.

**Comment by minniewong** [comment]
This is being addressed by GIBS: GIBS-37

**Comment by minniewong** [comment]
This is now being addressed here: ONEARTH-513

**Comment by minniewong** [comment]
@mcechini Will this be addressed in GIBS?

**Comment by mcechini** [comment]
Yes, but not until OnEarth 2.x is servicing request... so still another year away.  But I would like to see smarter caching someday

**Comment by hesam-oxe** [comment]
I can implement a client-side Service Worker that caches tiles 
locally using the Cache API — bypassing the server-side limitation 
entirely.

Benefits:
- Works regardless of server Cache-Control headers
- Reduces bandwidth for repeated tile requests
- Enables offline access to previously viewed areas
- Standard web technology — no external dependencies

PR up shortly. 🚀

**Comment by minniewong** [comment]
This ticket is OBE.
